bwmenu

```diff
@@ -158,28 +163,133 @@ clipboard-xsel-clear() {
   xsel --clipboard --delete
 }
 
-# Output the password
+# Copy the password
 # copy to clipboard and give the user feedback that the password is copied
 # $1: json item
-show_password() {
+copy_password() {
   pass=$(echo "$1" | jq -r '.login.password')
 
-  # send a notification for 5 seconds (-t 5000)
-  # not sure if icon will be present everywhere, /usr/share/icons is default icon location
-  notify-send "<b>$(echo "$1" | jq -r '.name')</b> copied" "${pass:0:4}****" -t 5000  -i /usr/share/icons/hicolor/64x64/apps/bitwarden.png
+  show_copy_notification "$1"
   echo -n "$pass" | clipboard-set
 
-  sleep 5
-  if [[ "$(clipboard-get)" == "$pass" ]]; then
-    clipboard-clear
+  if [[ $CLEAR -gt 0 ]]; then
+    sleep "$CLEAR"
+    if [[ "$(clipboard-get)" == "$pass" ]]; then
+      clipboard-clear
+    fi
+  fi
+}
+
+# Show notification about the password being copied.
+# $1: json item
+show_copy_notification() {
+  local title
+  local body=""
+
+  title="<b>$(echo "$1" | jq -r '.name')</b> copied"
+
+  if [[ $SHOW_PASSWORD == "yes" ]]; then
+    pass=$(echo "$1" | jq -r '.login.password')
+    body="${pass:0:4}****"
+  fi
+
+  if [[ $CLEAR -gt 0 ]]; then
+    body="$body<br>Will be cleared in ${CLEAR} seconds."
   fi
+  # send a notification for 5 seconds (-t 5000)
+  # not sure if icon will be present everywhere, /usr/share/icons is default icon location
+  notify-send "$title" "$body" -t 5000  -i /usr/share/icons/hicolor/64x64/apps/bitwarden.png
```

Comment:
The original intention was to show the notification as long as the password was on the clipboard.
```suggestion
  notify-send "$title" "$body" $timer  -i /usr/share/icons/hicolor/64x64/apps/bitwarden.png
```

Reply:
That makes sense.